scripts/perf_metrics.sh

```diff
@@ -0,0 +1,298 @@
+            ;;
+
+        $OUTPUT_PROM)
+            cat <<EOH
+
+# HELP node_cpu_usage_overall CPU usage overall
+# TYPE node_cpu_usage_overall gauge
+EOH
+            mpstat -o JSON 2 1 | jq -r '.sysstat.hosts[0].statistics[0]."cpu-load"[]|del(.cpu)|to_entries[]|"node_cpu_usage_overall{stat=\"\(.key)\"} \(.value)"'
+            ;;
+     esac
+}
+
+node_cpu_usage_process() {
+
+    # collect
+    cpu_usage_zeek=$(top -b -n1 |awk '$12 == "zeek" {print $9}')
+    cpu_usage_openvpn=$(top -b -n1 |awk '$12 == "openvpn" {print $9}')
+    cpu_usage_firemain=$(top -b -n1 -c|awk '$12 == "FireMain" {print $9}')
+    cpu_usage_firemon=$(top -b -n1 -c|awk '$12 == "FireMon" {print $9}')
+
+    # output
+    case $OUTPUT_FORMAT in
+    
+        $OUTPUT_SHELL)
+            cat <<EOS
+cpu_usage_zeek $cpu_usage_zeek
+cpu_usage_openvpn $cpu_usage_openvpn
+cpu_usage_firemain $cpu_usage_firemain
+cpu_usage_firemon $cpu_usage_firemon
+EOS
+            ;;
+
+        $OUTPUT_PROM)
+            cat <<EOP
+
+# HELP node_cpu_usage_process CPU usage of given process
+# TYPE node_cpu_usage_process gauge
+node_cpu_usage_overall{proc="zeek"} $cpu_usage_zeek
+node_cpu_usage_overall{proc="openvpn"} $cpu_usage_openvpn
+node_cpu_usage_overall{proc="firemain"} $cpu_usage_firemain
+node_cpu_usage_overall{proc="firemon"} $cpu_usage_firemon
+EOP
+            ;;
+    esac
+}
+
+node_network_xrate_bytes() {
+    case $(get_model) in
+        navy) intfs=$(ip --br l |awk '/(eth0|vpn_.*|tun_fwvpn)/ {print $1}') ;;
+        gold) intfs=$(ip --br l |awk '/(eth[0-3]|vpn_.*|tun_fwvpn)/ {print $1}') ;;
+        *) return ;;
+    esac
+
+    # output
+    case $OUTPUT_FORMAT in
+
+        $OUTPUT_SHELL)
+            for intf in $intfs; do
+                cat <<EOL
+network_xrate_${intf}_rx $(bmon  -p "${intf}" -o format:fmt='$(attr:rxrate:bytes)\n',format:quitafter=2 | tail -1)
+network_xrate_${intf}_tx $(bmon  -p "${intf}" -o format:fmt='$(attr:txrate:bytes)\n',format:quitafter=2 | tail -1)
+EOL
+            done
+            ;;
+
+        $OUTPUT_PROM)
+            cat <<EOH
+
+# HELP node_network_xrate_bytes Network transfer rate in bytes
+# TYPE node_network_xrate_bytes gauge
+EOH
+            for intf in $intfs; do
+                cat <<EOL
+node_network_xrate_bytes{if="$intf",tr="rx"} $(bmon  -p "${intf}" -o format:fmt='$(attr:rxrate:bytes)\n',format:quitafter=2 | tail -1)
+node_network_xrate_bytes{if="$intf",tr="tx"} $(bmon  -p "${intf}" -o format:fmt='$(attr:txrate:bytes)\n',format:quitafter=2 | tail -1)
+EOL
+            done
+            ;;
+    esac
+}
+
+node_ping_gateway_ms() {
+
+    # collect
+    gws=$(ip route | awk '$1 == "default" {print $3}')
+
+    # output
+    case $OUTPUT_FORMAT in
+
+        $OUTPUT_SHELL)
+            for gw in $gws; do
+                ping -nc 6 $gw | awk '/rtt/ {print $4}'| {
+                    IFS=/ read min avg max mdev
+                    gw_ip=$(echo $gw|tr '.' '_')
+                    cat <<EOL
+ping_${gw_ip}_min  $min
```

Comment:
there should be only one gw in the environment, it can also reduce the length of name

Reply:
Will take the first one if multiple gateways do exist.